pilot/pkg/bootstrap/server.go

```diff
@@ -431,13 +433,16 @@ func (s *Server) WaitUntilCompletion() {
 }
 
 // initSDSServer starts the SDS server
-func (s *Server) initSDSServer(args *PilotArgs) {
+func (s *Server) initSDSServer(args *PilotArgs) error {
 	if s.kubeClient != nil {
 		if !features.EnableXDSIdentityCheck {
 			// Make sure we have security
 			log.Warnf("skipping Kubernetes credential reader; PILOT_ENABLE_XDS_IDENTITY_CHECK must be set to true for this feature.")
 		} else {
-			sc := kubesecrets.NewMulticluster(s.kubeClient, s.clusterID, args.RegistryOptions.ClusterRegistriesNamespace)
+			sc, err := kubesecrets.NewMulticluster(s.kubeClient, s.clusterID, args.RegistryOptions.ClusterRegistriesNamespace)
```

Comment:
DANGER: an error in multicluster init should not result in Istiod failing to start. 
Not sure how errors are handled in the caller of initSDSServer or what can cause this to return an error, but it is almost always wrong to cause an total outage when one feature doesn't work ( and it can happen in unexpected ways - long after the config change is made ).

Reply:
how was this resolved?